src/Framework/Framework/Compilation/ControlTree/DataContextStack.cs

```diff
@@ -57,18 +57,17 @@ public IEnumerable<(int dataContextLevel, BindingExtensionParameter parameter)>
 
         public ImmutableArray<Delegate> GetAllBindingPropertyResolvers()
         {
-            // most likely, there aren't any so just quickly scan the link list
+            var builder = ImmutableArray.CreateBuilder<Delegate>();
+
             var c = this;
             while (c is {})
             {
-                if (!c.BindingPropertyResolvers.IsEmpty)
-                    break;
+                builder.AddRange(c.BindingPropertyResolvers);
```

Comment:
@exyi Is it possible that in the hierarchy (e.g. root -> child1) there is _child1_ with non-empty `BindingPropertyResolvers` and also _root_ with non-empty `BindingPropertyResolvers`? In that case this new implementation would be different than the original one - now `BindingPropertyResolvers` from both _chil1_ and _root_ are in the result where only `BindingPropertyResolvers` from _child1_ would be in the result in the original implementation.

Reply:
This case is possible, but I think that the behavior is still the same.

> only BindingPropertyResolvers from child1 would be in the result in the original implementation.

The EnumerableItems() enumerated itself and all ancestors, so it returned both root and child1. The SelectMany would then also include resolvers from both of them.

Reply:
You are right. Than it's ok.